merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/engine/SimulationEngine.java

```diff
@@ -463,26 +515,28 @@ public SimulationResults computeResults(
     final var simulatedActivities = new HashMap<ActivityInstanceId, SimulatedActivity>();
     final var unsimulatedActivities = new HashMap<ActivityInstanceId, SerializedActivity>();
     engine.tasks.forEach((task, state) -> {
-      final var directive = engine.taskDirective.get(task);
-      if (directive == null) return;
+      if (!taskInfo.isActivity(task)) return;
 
       final var activityId = taskToPlannedDirective.get(task.id());
 
       if (state instanceof ExecutionState.Terminated<?> e) {
+        final var inputAttributes = taskInfo.input().get(task.id());
+        final var outputAttributes = taskInfo.output().get(task.id());
+
         simulatedActivities.put(activityId, new SimulatedActivity(
-            directive.getType(),
-            directive.getArguments(),
+            inputAttributes.getTypeName(),
+            inputAttributes.getArguments(),
             startTime.plus(e.startOffset().in(Duration.MICROSECONDS), ChronoUnit.MICROS),
             e.joinOffset().minus(e.startOffset()),
```

Comment:
We could conceivably extract the start and end time of an activity from its input/output events, and remove that information from ExecutionState.

Reply:
Yes, I thought about that too. I'm worried that it's more of a *re-interpretation* of what it means for a task to have a duration, rather than a mere reflowing of information.

We have other things that don't fit into this paradigm (e.g. resource profiles), so I'm happy to punt on this here and deal with all of the misfits more holistically.

Reply:
(As a specific case, if the simulation is still in progress, how do we know *which event* is the last event? Maybe the task just hasn't emitted anything for a while!)